drivers/i2c/Kconfig.gd32

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2021 BrainCo Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_GD32_I2C := gd,gd32-i2c
+
+config I2C_GD32
+	bool "GD32 I2C driver"
+	depends on (SOC_FAMILY_GD32 || SOC_SERIES_GD32VF103)
+	default $(dt_compat_enabled,$(DT_COMPAT_GD32_I2C))
+	help
+	  Enables GigiDevice GD32 I2C driver
```

Comment:
So DT determines whether it is enabled/disabled by default, and then one can override this in Kconfig.

Is this standard practice?

I would think that whether a driver should be enabled/disabled should either be done in DT or in Kconfig.

Removing the prompt ("GD32 I2C driver"), would make this driver solely dependent on the DT configuration.

Removing line 9 would make this driver solely dependent on the Kconfig.

Reply:
Maybe just `def_bool $(dt_compat_enabled,$(DT_COMPAT_GD32_I2C))` then?

Reply:
If the driver control by CONFIG and DTS, we should use below step to enable it. 
```
dts:
    status = "okay";

prj.conf:
    CONFIG_I2C=y
    CONFIG_I2C_GD32=y
```

If use the dt_compat_enabled to fetch the default value from DTS, we can remove the CONFIG_I2C_GD32 from prj.conf.
```
dts:
    status = "okay";

prj.conf:
    CONFIG_I2C=y
```

From above examples, we can find there still use the CONFIG + DTS to enable the driver.

I prefer dt_compat_enabled to give the default value, it remove some duplicate work.

Reply:
> Maybe just `def_bool $(dt_compat_enabled,$(DT_COMPAT_GD32_I2C))` then?

done.

Reply:
Much better.